django/forms/models.py

```diff
@@ -52,6 +52,10 @@ def construct_instance(form, instance, fields=None, exclude=None):
             continue
         if exclude and f.name in exclude:
             continue
+        if not f.null and cleaned_data[f.name] is None:
+            # It's not the job of construct_instance to raise a possible ValueError
+            # here, let's delegate that to the validation step.
```

Comment:
Okay, I understand the idea. Suggested wording for this line: "here (assigning None to a non-nullable field), delegate..."

I've forgotten exactly what I wanted to check tonight, so I'll return to it Monday. 
When backporting, you could probably include 681df1aeafb30092430157f7977f713e1ce234ca too in order to minimize conflicts.